core/src/main/scala/kafka/tools/JmxTool.scala

```diff
@@ -122,7 +125,26 @@ object JmxTool extends Logging {
       else
         List(null)
 
-    val names = queries.flatMap((name: ObjectName) => mbsc.queryNames(name, null).asScala)
+    val hasPatternQueries = queries.exists((name: ObjectName) => name.isPattern)
+
+    retries = 0
+    var names : Iterable[ObjectName] = null
+    if (!hasPatternQueries) {
+      do {
+        if (names != null) {
+          System.err.println("Could not find all object names, retrying")
+          retries += 1
+          Thread.sleep(500)
```

Comment:
In general we've needed to increase these values. The 500 in the connection timeout was previously also an issue -- we had some tests that would exceed that. We sometimes see 5s of just JVM startup time before we ever even get a log message...